apiServer/taas/utils/k8sutils.go

```diff
@@ -117,3 +119,34 @@ func GetNodeStatus(c *gin.Context) {
 		"DegradedNodes":  degradedNodes,
 	})
 }
+
+// CreateVolumeSnapshotClass creates volume snapshot class
+func CreateVolumeSnapshotClass(c *gin.Context) {
+	log.Infof("Creating volume snapshot class")
+	var createVolumeSnapshotClassRequest struct {
+		VolumeSnapshotClassName      string `json:"volumeSnapshotClassName"`
+		Provisioner                  string `json:"provisioner"`
+		IsDefaultVolumeSnapshotClass bool   `json:"isDefaultVolumeSnapshotClass"`
+		DeletePolicy                 string `json:"deletePolicy"`
+	}
+	if !checkTorpedoInit(c) {
+		c.JSON(http.StatusInternalServerError, gin.H{
+			"error in init": fmt.Errorf("error in InitInstance()"),
+		})
+		return
+	}
+	if err := c.BindJSON(&createVolumeSnapshotClassRequest); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error for request": err.Error()})
+		return
+	}
+	if len(createVolumeSnapshotClassRequest.VolumeSnapshotClassName) == 0 || len(createVolumeSnapshotClassRequest.Provisioner) == 0 {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "volumesnapshotclass name  or provisioner cannot be empty", "vsc name": createVolumeSnapshotClassRequest.VolumeSnapshotClassName, "provisioner": createVolumeSnapshotClassRequest.Provisioner})
+		return
+	}
+	_, err := tests.Inst().S.CreateVolumeSnapshotClasses(createVolumeSnapshotClassRequest.VolumeSnapshotClassName, createVolumeSnapshotClassRequest.Provisioner, createVolumeSnapshotClassRequest.IsDefaultVolumeSnapshotClass, createVolumeSnapshotClassRequest.DeletePolicy)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+	c.JSON(http.StatusOK, gin.H{"message": fmt.Sprintf("VolumeSnapshotClass %v with provisioner %s created successfully", createVolumeSnapshotClassRequest.VolumeSnapshotClassName, createVolumeSnapshotClassRequest.Provisioner)})
```

Comment:
Let us also mention if we have set it as default or not

Reply:
done